examples/mcp/src/index.tsx

```diff
@@ -0,0 +1,120 @@
+import { gsx } from "gensx";
+import { Client } from "@modelcontextprotocol/sdk/client/index.js";
+import { StdioClientTransport } from "@modelcontextprotocol/sdk/client/stdio.js";
+import { GenerateText } from "@gensx/vercel-ai-sdk";
+import { openai } from "@ai-sdk/openai";
+import { tool, ToolSet } from "ai";
+import { z } from "zod";
+
+
+interface MCPServerResult {
+  client: Client;
+  availableTools: any;
+}
+interface MCPServerProps {
+  clientName: string;
+  clientVersion: string;
+  serverCommand: string;
+  serverArgs: string[];
+}
+
+const MCPServer = gsx.Component<MCPServerProps, MCPServerResult>(
```

Comment:
I think this should be a Provider, and makes the tools available in a context.

Reply:
I'd be curious to see what that code looks like. Haven't created a provider yet. 

In terms of provider vs component, I wonder what the rule of thumb should be. I could go either way here. It would be the client you would be passing down which has a set of functions on it to take actions (which actually doesn't work outside of the component where's it defined)

Reply:
I agree with Jeremy in this instance. 

You don't want to have to wrap every ChatCompletion component with an MCPServer component in a real workflow. Makes much more sense to do it at the top of the workflow and then access the tools with `useContext` in any of the subcalls within a tree where you need them.